packages/vitest/src/node/cli.ts

```diff
@@ -128,7 +129,7 @@ function normalizeCliOptions(argv: CliOptions): CliOptions {
 async function start(mode: VitestRunMode, cliFilters: string[], options: CliOptions): Promise<Vitest | undefined> {
   try {
     const ctx = await startVitest(mode, cliFilters.map(normalize), normalizeCliOptions(options))
-    if (!ctx?.config.watch)
+    if (!(ctx?.config.watch || ctx?.config.browser))
```

Comment:
Why do we need to always add `config.browser` check everywhere now? It should be unified. If it's not possible to run them together for now, then `watch` should be set to `false`

Reply:
The browser mode behaves differently, we always need the server to be running, so we cannot just close the context right away.

- In watch mode we have the server, so no additionals, we just stick to that server.
- In run mode, the server gets closed soon, so in the previous PR also, the run mode didn't work properly. Now what we do is that we let the server run and not close it right away, once we run the tests in browser, on onFinished, we close the server now.

Reply:
There should be a unified way to do this, we can move the check into it's own method